media/js/firefox/firstrun/membership.js

```diff
@@ -0,0 +1,214 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function (Mozilla) {
+    'use strict';
+
+    var client = Mozilla.Client;
+    var _supportsWebAnimations = 'animate' in document.createElement('div');
+    var _supportsPromises = 'Promise' in window;
+    var cutsTheMustard = _supportsWebAnimations && _supportsPromises;
+    var $slideShow = $('.carousel');
+    var button = document.querySelectorAll('.path-selector ul > li > button');
+    var step1 = document.querySelector('.step-1');
+    var step2 = document.querySelector('.step-2');
+    var laterLink = document.querySelector('a.later');
+
+    function initSlideShow() {
+        $slideShow.cycle({
+            fx: 'scrollHorz',
+            log: false,
+            slides: '> .carousel-item',
+            pager: '> .carousel-pager',
+            pauseOnHover: true,
+            speed: 620,
+            timeout: 6000,
+            delay: 5000
+        });
+
+        $slideShow.on('cycle-pager-activated', function(event, opts) {
+            var slide;
+            switch(opts.currSlide) {
+            case 0:
+                slide = 'Independence';
+                break;
+            case 1:
+                slide = 'Privacy';
+                break;
+            case 2:
+                slide = 'Technology';
+                break;
+            }
+
+            if (slide) {
+                window.dataLayer.push({
+                    'event': 'in-page-interaction',
+                    'eAction': 'Carousel',
+                    'eLabel': slide
+                });
+            }
+        });
+    }
+
+    function initAccountsForm() {
+        if (client.isFirefoxDesktop) {
+            // initialize FxA iframe form
+            client.getFirefoxDetails(function(data) {
+                Mozilla.FxaIframe.init({
+                    distribution: data.distribution,
+                    gaEventName: 'firstrun-fxa'
+                });
+            });
+        }
+    }
+
+    // replaces utm_campaign param in FxA iframe data-src with data-id value from button click.
+    function updateUTMParam(id) {
+        var utmCampaign = id ? encodeURIComponent(id) : 'fxa-embedded-form';
+        var iframe = document.getElementById('fxa');
+        var dataSrc = iframe.getAttribute('data-src');
+        iframe.setAttribute('data-src', dataSrc.replace('utm_campaign=fxa-embedded-form', 'utm_campaign=' + utmCampaign));
```

Comment:
As this is going to run in recent versions of Firefox only, any reason not to do the above as follows:

```
var dataSrc = iframe.dataset.src
iframe.dataset.src = dataSrc.replace('utm_campaign=fxa-embedded-form', 'utm_campaign=' + utmCampaign);
```